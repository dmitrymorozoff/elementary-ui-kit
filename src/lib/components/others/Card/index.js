import React from "react";
import styled from "styled-components";
import { PropTypes } from "prop-types";

export const Card = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    min-width: 275px;
    min-height: 260px;
    width: ${props => props.width};
    height: ${props => props.height};
    box-shadow: 0 8px 25px -6px #f2f2f6;
    position: relative;
    margin: ${props => (props.withImage ? "45px 10px 0 10px" : "10px")};
    border: 1px solid #f2f2f5;
    box-sizing: border-box;
    box-shadow: 0 15px 20px -15px rgba(0, 0, 0, 0.15), 0 55px 50px -35px rgba(0, 0, 0, 0.15);
    padding: 15px;
    transition: 0.5s;
    cursor: ${props => (props.hover ? "pointer" : "default")};
    &:hover {
        box-shadow: ${props =>
            props.hover
                ? "0 15px 10px -15px rgba(0, 0, 0, 0.15), 0 35px 50px -35px rgba(0, 0, 0, 0.15)"
                : "0 15px 20px -15px rgba(0, 0, 0, 0.15), 0 55px 50px -35px rgba(0, 0, 0, 0.15)"};
    }
`;

Card.defaultProps = {
    width: "275px",
    height: "260px",
};

Card.propTypes = {
    children: PropTypes.node,
    width: PropTypes.string,
    height: PropTypes.string,
    hover: PropTypes.bool,
    withImage: PropTypes.bool,
};
