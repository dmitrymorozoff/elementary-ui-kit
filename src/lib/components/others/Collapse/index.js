import React from "react";
import { PropTypes } from "prop-types";
import styled from "styled-components";
import { CollapseTitle } from "./components/CollapseTitle";
import { CollapseContent } from "./components/CollapseContent";

const CollapseWrapper = styled.div`
    font-family: "Roboto", sans-serif;
    transition: 0.2s;
    width: ${props => (props.fullWidth ? "100%" : "500px")};
    background-color: #fff;
    border-left: 3px solid ${props => (props.isOpen ? props.color : "#fff")};
    border: 1px solid #f2f2f5;
    box-shadow: 0 15px 20px -15px rgba(0, 0, 0, 0.1), 0 55px 50px -35px rgba(0, 0, 0, 0.1);
    &:hover {
        border: 1px solid ${props => (props.isOpen ? "#f2f2f5" : props.color)};
    }
`;

export class Collapse extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
        };
    }

    getChildrens() {
        const { children, color } = this.props;
        const { isOpen } = this.state;
        if (!Array.isArray(children)) {
            return children;
        }
        return children.map((child, index) => {
            return index === 0 ? (
                <CollapseTitle
                    key={index.toString()}
                    isOpen={isOpen}
                    color={color}
                    onClick={this.toggleCollapse.bind(this)}
                >
                    {child.props.children}
                </CollapseTitle>
            ) : (
                <CollapseContent key={index.toString()} isOpen={isOpen}>
                    {child.props.children}
                </CollapseContent>
            );
        });
    }

    toggleCollapse() {
        const { onChange } = this.props;
        const newIsOpenValue = !this.state.isOpen;
        this.setState({
            isOpen: newIsOpenValue,
        });
        onChange(newIsOpenValue);
    }

    render() {
        const { fullWidth, color } = this.props;
        const { isOpen } = this.state;
        return (
            <CollapseWrapper fullWidth={fullWidth} isOpen={isOpen} color={color}>
                {this.getChildrens()}
            </CollapseWrapper>
        );
    }
}

Collapse.defaultProps = {
    fullWidth: false,
    color: "#3a77f8",
    onChange: () => {},
};

Collapse.propTypes = {
    onChange: PropTypes.func,
    children: PropTypes.node,
    fullWidth: PropTypes.bool,
    color: PropTypes.string,
};
