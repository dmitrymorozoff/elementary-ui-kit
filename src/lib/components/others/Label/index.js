import styled from "styled-components";
import { PropTypes } from "prop-types";

const defaultColors = props => {
    if (props.type === "primary") return props.theme.type.primary;
    if (props.type === "success") return props.theme.type.success;
    if (props.type === "warning") return props.theme.type.warning;
    if (props.type === "danger") return props.theme.type.danger;
    return "transparent";
};

export const Label = styled.div`
    display: inline-block;
    border: 1px solid ${defaultColors};
    width: auto;
    font-family: "Roboto", sans-serif;
    font-size: 12px;
    background-color: ${defaultColors};
    padding: 4px 12px;
    text-transform: uppercase;
    box-shadow: 0 4px 20px -6px ${defaultColors};
    color: ${props => {
        if (props.type) {
            return "#fff";
        }
    }};
`;

Label.defaultProps = {
    theme: {
        type: {
            default: "#d8dadd",
            primary: "#3A77F8",
            success: "#4FD496",
            warning: "#FAAF27",
            danger: "#FE0C0D",
        },
    },
};

Label.propTypes = {
    theme: PropTypes.object,
    type: PropTypes.oneOf(["default", "primary", "success", "warning", "danger"]),
};
