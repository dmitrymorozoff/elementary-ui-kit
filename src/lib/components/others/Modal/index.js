import React from "react";
import styled from "styled-components";
import { PropTypes } from "prop-types";

const Outer = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    overflow: auto;
    height: 100%;
    z-index: 1;
    transition: 0.35s;
    opacity: ${props => (props.open ? "1" : "0")};
    visibility: ${props => (props.open ? "visible" : "hidden")};
`;

const Overlay = styled.div`
    position: fixed;
    display: flex;
    justify-content: center;
    align-items: center;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.045);
`;

const ModalItem = styled.div`
    position: relative;
    width: 500px;
    padding: 20px;
    box-sizing: border-box;
    background-color: #fff;
    border-radius: 3;
    z-index: 2;
    border: 1px solid #f2f2f5;
    text-align: left;
    box-shadow: rgba(0, 0, 0, 0.15) 0px 15px 20px -15px,
        rgba(0, 0, 0, 0.15) 0px 55px 50px -35px;
    transition: 0.35s;
    transform: ${props => (props.open ? "scale(1)" : "scale(1.2)")};
`;

export class Modal extends React.Component {
    constructor(props) {
        super(props);
        this.overlay = React.createRef();
        this.handleKeyUp = this.handleKeyUp.bind(this);
        this.handleClose = this.handleClose.bind(this);
    }

    componentDidMount() {
        window.addEventListener("keydown", this.handleKeyUp, true);
    }

    componentWillUnmount() {
        window.removeEventListener("keydown", this.handleKeyUp, true);
    }

    handleKeyUp(event) {
        const { closeOnEsc, onClose } = this.props;
        const KEY_CODE_ESC = 27;
        if (closeOnEsc && event.keyCode === KEY_CODE_ESC) {
            onClose();
        }
    }

    handleClose(event) {
        const { onClose } = this.props;
        if (this.overlay.current !== event.target) {
            return;
        }
        onClose();
    }

    render() {
        const { open, children, className } = this.props;
        return (
            <Outer open={open} tabIndex="-1" className={className}>
                <Overlay onClick={this.handleClose} innerRef={this.overlay}>
                    <ModalItem open={open}>{children}</ModalItem>
                </Overlay>
            </Outer>
        );
    }
}

Modal.defaultProps = {
    open: false,
    closeOnEsc: false,
};

Modal.propTypes = {
    children: PropTypes.node,
    onClose: PropTypes.func.isRequired,
    open: PropTypes.bool.isRequired,
    closeOnEsc: PropTypes.bool,
    className: PropTypes.string,
};
