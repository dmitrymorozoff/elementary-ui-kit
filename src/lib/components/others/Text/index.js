import React from "react";
import styled from "styled-components";
import { PropTypes } from "prop-types";

export const Text = styled.p`
    font-family: "Lato", sans-serif;
    line-height: 1.5;
    font-size: ${props => props.fontSize};
    margin: 0;
    margin-bottom: ${props => (props.gutterBottom ? "8px" : 0)};
`;

Text.defaultProps = {
    fontSize: "14px",
    gutterBottom: false,
};

Text.propTypes = {
    fontSize: PropTypes.string,
    gutterBottom: PropTypes.bool,
};
