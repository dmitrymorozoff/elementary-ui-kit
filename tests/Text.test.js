import React from "react";
import { Text } from "../src/lib/components/others/Text";

describe("<Text/>", () => {
    it("should render a default text", () => {
        const component = shallow(
            <Text>
                They live almost exclusively in the Southern Hemisphere, with only one species, the Galapagos penguin,
                found north of the equator.
            </Text>,
        );
        expect(component).toMatchSnapshot();
    });
});
