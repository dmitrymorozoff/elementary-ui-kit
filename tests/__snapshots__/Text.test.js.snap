// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> should render a default text 1`] = `
<p
  className="sc-gqjmRU Abaiz"
  fontSize="14px"
>
  They live almost exclusively in the Southern Hemisphere, with only one species, the Galapagos penguin, found north of the equator.
</p>
`;
